Extension methods can be exported by name (#10274)
enso-org/enso: docs/syntax/functions.md

## Methods

Enso makes a distinction between functions and methods. In Enso, a method is a
function where the first argument (known as the `self` argument) is associated
with a given atom. Methods are dispatched dynamically based on the type of the
`self` argument, while functions are not.

Methods can be defined in Enso in two ways:

1. **In the Body of a Type:** A function defined in the body of a `type`
   definition is automatically converted to a method on all the atoms defined in

>    the body of that type definition. It is call an _instance method_. If the

Reviewer: ```suggestion
   the body of that type definition. If the function has `self` parameter, it is called an _instance method_. If the
```

>    function defined in the body of a type does not have `self` parameter, it is

Reviewer: ```suggestion
   function does not have `self` parameter, it is called
```

2. **As an Extension Method:** A function defined _explicitly_ on an atom counts
   as an extension method on that atom. It can be defined on a typeset to apply

>    to all the atoms within that typeset. An _extension_ method can be _static_

Reviewer: `typeset`!? Looks like a relict of some old terminology. Also _"function defined _explicitly_ on an atom"_ doesn't seem to make any sense anymore. In both cases we should probably talk about **type**.

Author: Yes, that is a relic. Updated in 46df67bce940f3b56ced66ebd04ffd3b78239c3a

   or _instance_, depending on whether the `self` argument is present or not.

```ruby
Number.floor self = case self of
    Integer -> ...
    ...
```

3. **As a Function with an Explicit `self` Argument:** A function defined with
   the type of the `self` argument specified to be a type.

```ruby

> floor (self : Number) = case this of

Reviewer: ```suggestion
floor (self : Number) = case self of
```
